src/components/Steps/Search/index.js

```diff
@@ -31,11 +32,13 @@ export class Search extends Component {
         geocodeByAddress(selected)
             .then(results => getLatLng(results[0]))
             .then(({ lat, lng }) => {
-                this.setState({
-                    address: selected,
+                this.props.saveLocalisation({
                     latitude: lat,
                     longitude: lng,
                 });
+                this.setState({
+                    address: selected,
```

Comment:
Are You sure that we'll not need address in redux ? I can imagine that we can show the name of job/school on the map. Then You could push it to redux and remove state at all :)